### ISSUE TITLE
Wildcard imports are not ignored in __init__.py files

### ISSUE THREAD

**Issue by Hornwitser** [PyCQA/pylint#2430]
As discussed in #2026 and implemented in #2084, the code to ignore wildcard imports in `__init__.py` files doesn't work.

### Steps to reproduce
1.  Create an `__init__.py` file in a `pkg` folder, with the content
```python
"""pkg"""
from json import *
```
2. run `pylint pkg` from the folder containing the `pkg` folder

### Current behavior
Reports warning `pkg/__init__.py:2:0: W0401: Wildcard import json (wildcard-import)` due to to the skip check expecting the [module node's name attribute to be equal to `"__init__"`](https://github.com/PyCQA/pylint/pull/2084/commits/62736b4fd658a8a79dca40a1285080e408140054#diff-cb70a608727227ba4c747028843ca72bR791) while the actual value is the package name (i.e. `"pkg"`.)

### Expected behavior
Wildcard imports are ignored from `__init__.py` files.

### pylint --version output
```
pylint --version
pylint 2.1.1
astroid 2.0.1
Python 3.7.0 (default, Jul 15 2018, 10:44:58)
[GCC 8.1.1 20180531]
```

**Comment by PCManticore** [comment]
Thanks for creating an issue!

**Comment by Hornwitser** [comment]
Looking at the [unit test added](https://github.com/PyCQA/pylint/pull/2084/commits/0190db7c1e30bd997fa8e08831a3fddd1de7da0e#diff-d9cc6edcff66800a456aaf912471fa19R101) it seems the code it uses to create a module object is overly verbose (and wrong).
```python
here = os.path.abspath(os.path.dirname(__file__))
path = os.path.join(
    here, 'regrtest_data', 'init_wildcard', '__init__.py')
with open(path) as stream:
    data = stream.read()
module = astroid.parse(data, module_name='__init__', path=path)
```
It would be much simpler to use `ast_from_module_name`.
```python
# Have this at the top of the module or somewhere accessible
REGRTEST_DATA = os.path.join(os.path.dirname(__file__), 'regrtest_data', '')

# Then in the unit test
module = astroid.MANAGER.ast_from_module_name('init_wildcard', REGRTEST_DATA)
```
